### PR TITLE
clear input text on escape key #680

### PR DESCRIPTION
**Pull request by gaskar**
**What**: Fixed issue concerning escape key handling for comboboxes from w3 standards
https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html

Fixes https://github.com/downshift-js/downshift/issues/680

**How**: Set inputValue to empty string on escape key handler, before calling reset method.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

File: src/downshift.js [hunk 1]
```diff
@@ -610,7 +610,10 @@ class Downshift extends Component {
 
     Escape(event) {
       event.preventDefault()
-      this.reset({type: stateChangeTypes.keyDownEscape})
+      this.reset({
+        type: stateChangeTypes.keyDownEscape,
+        inputValue: '',
```
Comment (silviuavram) [inline src/downshift.js]: I have been thinking about this. if we clear the input, we should also remove the selected item, if any. in the ARIA docs, the whole point of the combobox is to fill this input with a string value, doesn't matter what the value you compute it from is. if ARIA says the input is cleared at Escape, then there is not point in having a selected item.

also, I think our reset function needs to clear the selected item by default. but unfortunately right now it's being used in our blur handles. ARIA says that on blur, if there is a highlighted item, it should be selected. I created an issue for this https://github.com/downshift-js/downshift/issues/679.

that being said, if you agree, we can add selectedItem as null here, update the documentation on the Escape handler, and then we can start working on having selectedItem: null in reset function and then selecting the item on blur, thus removing some uses of the reset function. in a separate PR. what do you think?

Comment (gaskar) [reply src/downshift.js]: @silviuavram agree, sorry I didn't see your comment here, will do as you suggested and update PR.

Comment (gaskar) [reply src/downshift.js]: @silviuavram can you revisit this PR?

**Pull request merged**